hasher-matcher-actioner/hmalib/lambdas/hashing.py

```diff
@@ -71,30 +78,46 @@ def lambda_handler(event, context):
         if message.get("Event") == "s3:TestEvent":
             continue
 
+        media_to_process: t.List[t.Union[S3ImageSubmission, URLSubmissionMessage]] = []
+
         if URLSubmissionMessage.could_be(message):
-            submission_message = URLSubmissionMessage.from_sqs_message(message)
+            media_to_process.append(URLSubmissionMessage.from_sqs_message(message))
+        elif S3ImageSubmissionBatchMessage.could_be(message):
```

Comment:
same (not blocking) nit about could_be from earlier PRs: I find this flow harder to follow and refactoring will be ~tricker than having, could_be and from be linked in Message class and treated as a single call here.

Reply:
I forget. Are you suggesting instead of `could_be`, we simply try to deserialize and use a None value as our hint that it was not successful?

eg. 
```python
message = ClassA.from_sqs_message(raw_message)
if message is None:
  message = ClassB.from_sqs_message(raw_message)
if message is None:
  message = ClassC.from_sqs_message(raw_message)
```

Reply:
ignore-able: effectively, maybe with prefix `.try_from_sqs_message(...)` and not as a hint that waterfalls just as a quick check for each:

```python
if m := ClassA.try_from_sqs_message(raw_message): media_to_process.append(m)
if m := ClassB.try_from_sqs_message(raw_message): media_to_process.append(m)
if m := ClassC.try_from_sqs_message(raw_message): media_to_process.append(m)
```
or even loop through possible message types.